lib/docker2aci.go

```diff
@@ -167,24 +166,39 @@ func SquashLayers(images []acirenderer.Image, aciRegistry acirenderer.ACIRegistr
 	}
 
 	squashedFilename := getSquashedFilename(parsedDockerURL)
-	squashedImagePath := path.Join(outputDir, squashedFilename)
+	squashedImagePath := filepath.Join(outputDir, squashedFilename)
 
-	squashedImageFile, err := os.Create(squashedImagePath)
+	squashedTempFile, err := ioutil.TempFile(outputDir, "docker2aci-squashedFile-")
 	if err != nil {
 		return "", err
 	}
-	defer squashedImageFile.Close()
+	defer func() {
+		errClose := squashedTempFile.Close()
+		if err != nil {
+			// remove temp file on error
+			errRemove := os.Remove(squashedTempFile.Name())
+			if errRemove != nil {
+				err = errRemove
```

Comment:
> Maybe we should return the "real" error instead of the Remove error? Not sure what's better.

I've got the same doubts. I'll ignore the Close and the Remove errors to not mask the "real" one. (for example, Close() errors are often ignored: `defer file.Close()`)